pkg/build/builder/docker.go

```diff
@@ -154,7 +160,15 @@ func (d *DockerBuilder) addBuildParameters(dir string) error {
 
 	var newFileData string
 	if d.build.Parameters.Strategy.DockerStrategy.BaseImage != "" {
-		newFileData = imageRegex.ReplaceAllLiteralString(string(fileData), fmt.Sprintf("FROM %s", d.build.Parameters.Strategy.DockerStrategy.BaseImage))
+		cmd := dockercmd.From
+		replaceStr := fmt.Sprintf("%s %s", strings.ToUpper(cmd), d.build.Parameters.Strategy.DockerStrategy.BaseImage)
+		newFileData, err = replaceValidCmd(cmd, replaceStr, fileData)
+		if err != nil {
+			// replaceValidCmd always returns a complete Dockerfile
+			// so the question is: ignore this error and fallback
+			// to using the original Dockerfile or return err ?
```

Comment:
I think we should err out..if we failed to do the replace, then we're not building what we're supposed to be building.